extensions/agroal/deployment/src/test/java/io/quarkus/agroal/test/DefaultDataSourceConfigTest.java

```diff
@@ -59,6 +59,15 @@ private static void testDataSource(AgroalDataSource dataSource, String username,
         assertTrue(agroalConnectionFactoryConfiguration.trackJdbcResources());
         assertTrue(dataSource.getConfiguration().metricsEnabled());
         assertEquals(newConnectionSql, agroalConnectionFactoryConfiguration.initialSql());
+        assertTrue(agroalConnectionFactoryConfiguration.jdbcProperties().containsKey("socketFactory"));
+        assertEquals(agroalConnectionFactoryConfiguration.jdbcProperties().getProperty("socketFactory"),
+                "pass socket factory provider as jdbc property");
+        assertTrue(agroalConnectionFactoryConfiguration.jdbcProperties().containsKey("extraProperty1"));
+        assertEquals(agroalConnectionFactoryConfiguration.jdbcProperties().getProperty("extraProperty1"),
+                "extraProperty1Value");
+        assertTrue(agroalConnectionFactoryConfiguration.jdbcProperties().containsKey("extraProperty2"));
+        assertEquals(agroalConnectionFactoryConfiguration.jdbcProperties().getProperty("extraProperty2"),
+                "extraProperty2Value");
         try (Connection connection = dataSource.getConnection()) {
         }
```

Comment:
You can use AssertJ here (add org.assertj:assetj-core in test scope in the pom.xml for this project):

```java
        assertThat(agroalConnectionFactoryConfiguration.jdbcProperties())
                .contains(entry("socketFactory", "pass socket factory provider as jdbc property"),
                        entry("extraProperty1", "extraProperty1Value"),
                        entry("extraProperty2", "extraProperty2Value"));
        assertThatCode(() -> dataSource.getConnection().close()).doesNotThrowAnyException();

```

Reply:
I like your suggestion but I will wait for a second opinion if we should add the extra dependency in the pom or not

Reply:
It's fine to use AssertJ. It is used in other extensions and it's in test scope, so that's not a problem :)